app/uk/gov/hmrc/tariffclassificationfrontend/models/request/AuthenticatedRequest.scala

```diff
@@ -17,6 +17,24 @@
 package uk.gov.hmrc.tariffclassificationfrontend.models.request
 
 import play.api.mvc.{Request, WrappedRequest}
-import uk.gov.hmrc.tariffclassificationfrontend.models.Operator
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.AccessType.{AccessType, READ_WRITE}
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, Operator}
 
-case class AuthenticatedRequest[A](operator: Operator, request: Request[A]) extends WrappedRequest[A](request)
+class AuthenticatedRequest[A](_operator: Operator, _request: Request[A], _accessType: AccessType = READ_WRITE)
+                  extends WrappedRequest[A](_request) {
+
+  val operator : Operator = _operator
+  val request  : Request[A] = _request
+
+  val hasWritePermission: Boolean = _accessType == AccessType.READ_WRITE || operator.manager
+}
+
+class AuthenticatedCaseRequest[A](operator: Operator, request: Request[A], accessType: AccessType = READ_WRITE, _c: Case)
+                  extends AuthenticatedRequest[A] (operator , request, accessType){
+  val c : Case = _c
```

Comment:
Can we call this something more descriptive than `c` please :) e.g. `case` with back ticks